### ISSUE TITLE
feat: use shorebird's dart fork always

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#2105]
Right now we have a funny setup, where by `flutter` when invoked from `shorebird` will talk to downloads.shorebird.dev, which runs artifact_proxy, and will re-write some requests to pull shorebird artifacts rather than Google/flutter ones.

We replace many things, including gen_snapshot.  However we do not replace the dart-sdk itself.

This causes kernel sdk_hash mismatches, so we've currently disabled kernel sdk_hash checks.

The correct solution to get around this is to vend our own Dart, rather than using Google's.   This will require doing a few things:
1. Making sure that when we build our Dart fork, it works correctly with all our #defines turned off.
2. Set up build_engine and builders for all the various flavors of Dart we need to vend.
3. Configure the artifact proxy to point to our uploaded Dart.

**Comment by felangel** [comment]
I think we should be able to close this now but leaving it open until we re enable the kernel sdk hash checks.

**Comment by eseidel** [comment]
Yeah, very small change to complete this.  We just change sdk_args.gni to reenable the check.

**Comment by eseidel** [comment]
OK, I made a ton of progress on this this week, but have ultimately decided to back off and not complete this work for now.

The blocker that stuck me was https://github.com/flutter/flutter/issues/164531.  Which will require us to not use --no-dart-precompiled-sdk for x64 mac builds (or maybe to hack the build.gn to modify the host_arch to match mac_cpu before calling out to dart build rules? Not even sure if that's possible).

I tried to work around https://github.com/flutter/flutter/issues/164531 by building the dart-sdk separately for x64 "create_platform_sdk" target, but ran into compile errors there, I think because SHOREBIRD_SNAPSHOT changes the snapshot for JIT builds not just AOT builds (we could be more careful not to) which caused JIT behavior to break/crash.  I was going to take the compiled dart SDK and swap it into the `prebuilts` folder, overriding what DEPS had downloaded for mac-x64/dart-sdk.

But in the end I wasn't sure that even after that I would have things all fixed, so I'm putting this down for now and will pick it up later.  Eventually we need to make it possible to build/run Shorebird's Dart everywhere and then update our bots to build all the flavors of Dart we need.

Right now were relying on the fact that Shorebird Dart snapshots and upstream Dart snapshots are similar-enough to not crash in most cases, but customers have seen crashes.  Right now we just disable the snapshot version hash checks, which isn't great, but has gotten us this far over 2 years and will have to hold out for a few more months until someone can get back to this.

**Comment by eseidel** [comment]
Changes I made before giving up: https://github.com/shorebirdtech/_build_engine/compare/6df72e18a27a295b6931567e96cbe96c3915f39e..main

**Comment by eseidel** [comment]
font-subset changes to artifact_proxy I couldn't land: https://github.com/shorebirdtech/shorebird/commit/6f1a8728d0078a8dc6cf9edf5125ee1d723c1286

**Comment by eseidel** [comment]
Tracking the build-pipeline implementation work for step 2 internally. Will surface back here as the public-facing pieces land.
